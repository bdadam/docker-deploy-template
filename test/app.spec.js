const request = require('supertest');
const app = require('../src/app');

describe('GET /', () => {
    var response;
    beforeAll(async () => {
        response = await request(app).get('/');
    });

    it('200 OK', () => {
        expect(response.status).toEqual(200);
    });

    it('Content-Type: text/html', () => {
        expect(response.type).toEqual('text/html');
    });

    it('HTML Snapshot', () => {
        expect(response.text).toMatchSnapshot();
    });

    it('Headers', () => {
        const hds = Object.assign({}, response.headers, { date: 'dontcare', 'server-timing': 'dontcare' });
        expect(hds).toMatchSnapshot();
    });
});