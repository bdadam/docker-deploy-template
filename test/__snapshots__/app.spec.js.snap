// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET / HTML Snapshot 1`] = `
<html>
  
  <head>
    <title>Test Title This is</title>
  </head>
  
  <body>
    <h1>Test Title</h1>
    <form id="dataform">
      <textarea name="text" rows="6" cols="80"></textarea>
      <button type="submit">Submit</button>
    </form>
    <script src="/static/main.js?qweq12324"></script>
  </body>

</html>
`;
